### PR TITLE
Swift lint

### PR DESCRIPTION
**Pull request by ZaweSK**
- initialize gitignore - inspired by https://github.com/github/gitignore/blob/master/Swift.gitignore
- initialize pod file, add SwiftLint pod, create a workspace
- update code style to satisfy SwiftLint rules
- update project group structure

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
```
Comment (Pacek) [inline Podfile]: platform version should be specified and match the project deployment target

File: AcademyApp/AppDelegate.swift [hunk 2]
```diff
@@ -13,34 +13,26 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+//    }
+//
+//    func applicationWillEnterForeground(_ application: UIApplication) {
+//    }
+//
+//    func applicationDidBecomeActive(_ application: UIApplication) {
+//    }
+//
+//    func applicationWillTerminate(_ application: UIApplication) {
+//    }
```
Comment (Pacek) [inline AcademyApp/AppDelegate.swift]: delete it if you don't need it, don't commit commented code

**Comment by Pacek** [comment]
- shows me one warning at the end of LoginViewController - Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
- generally in classes/structs properties should be on one place and first followed by init/deinit and methods afterwards - so the order for LoginViewController should be: properties/IBOutlets, init/deinit, lifecycle, actions, methods
- something is off with the swiftlint setup - you have private_action and private_outlet rules on but you have all outlets and actions internal and it's not showing up warnings

File: Podfile (hunk 1 above)
Comment (cejanen) [reply Podfile]: +1

Comment (ZaweSK) [reply Podfile]: Just wondering ... Is it necessary to also do **pod update** after the ios platform in podfile has been updated?

File: AcademyApp/AppDelegate.swift (hunk 2 above)
Comment (cejanen) [reply AcademyApp/AppDelegate.swift]: +1

Comment (ZaweSK) [reply AcademyApp/AppDelegate.swift]: Removed unnecessary methods from AppDelegate.

**Comment by ZaweSK** [comment]
>     * shows me one warning at the end of LoginViewController - Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
> 
>     * generally in classes/structs properties should be on one place and first followed by init/deinit and methods afterwards - so the order for LoginViewController should be: properties/IBOutlets, init/deinit, lifecycle, actions, methods
> 
>     * something is off with the swiftlint setup - you have private_action and private_outlet rules on but you have all outlets and actions internal and it's not showing up warnings

*  class organization
Would you say it's ok to keep computed properties, which are used for specific code parts (like UI Setup Methods) grouped by extension within that extension ? 

image

**Review by cejanen — COMMENTED** [review]
detail - podfile format - missing left space + I would advice keeping pods in commented sections like
'# ui'
'# networking'
'# swiftlint'

why is this rule disabled?
disabled_rules:
  - trailing_whitespace

I would suggest name run script in project settings to swiftlint (there can be more run scripts) so it obvious

About what @Pacek wrote and it doesnt show warnings for private IBOutlets also this should be warned
passwordText.count > 0 and needs passwordText.isEmpty instead
or passwordTextField.isSecureTextEntry = !passwordTextField.isSecureTextEntry should be 
.toggle etc

**Comment by cejanen** [comment]
sry for formatting, it was 
'#ui' etc

File: Podfile (hunk 1 above)
Comment (cejanen) [reply Podfile]: Nope but its good to lock pods at some version so ur sure when somebody does pod install/pod update it doesnt break build

Comment (cejanen) [reply Podfile]: Pod install is enough, check difference between pod install and pod update

Comment (ZaweSK) [reply Podfile]: Yep, good point.

Comment (ZaweSK) [reply Podfile]: Locked pod version,  specified deployment target

**Comment by Pacek** [comment]
> > ```
> > * shows me one warning at the end of LoginViewController - Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
> > 
> > * generally in classes/structs properties should be on one place and first followed by init/deinit and methods afterwards - so the order for LoginViewController should be: properties/IBOutlets, init/deinit, lifecycle, actions, methods
> > 
> > * something is off with the swiftlint setup - you have private_action and private_outlet rules on but you have all outlets and actions internal and it's not showing up warnings
> > ```
> 
> 
> 
> * class organization
>   Would you say it's ok to keep computed properties, which are used for specific code parts (like UI Setup Methods) grouped by extension within that extension ?
> 
> 
> 
> image

personally I would have all properties on top and make them all private unless you need them from outside

**Comment by ZaweSK** [comment]
About swiftLint... I am using .yaml file posted on strv github. I am not quite sure about the broken rules you stated. We can check together once you'll be in office.

File: .swiftlint.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+# SwiftLint configuration file commonly used in STRV (v 0.1.2)
+#
+# Feel free to send pull request or suggest improvements!
+#
+
+
+#
+# Rule identifiers to exclude from running.
+#
+disabled_rules:
```
Comment (cejanen) [inline .swiftlint.yml:11]: why is this rule disabled?

File: AcademyApp/ViewControllers/LoginViewController.swift [hunk 4]
```diff
@@ -80,151 +85,131 @@ extension LoginViewController : UITextFieldDelegate
...
+
     // MARK: - UI Setup Methods
-    
-    func setupUI(){
+
+    func setupUI() {
         backgroundColorSetup()
         textFieldsSetup()
-        showLabelSetup()
+        showButtonSetup()
```
Comment (cejanen) [inline AcademyApp/ViewControllers/LoginViewController.swift]: would suggest have different name... its more like setup some kind of button, showButtonSetup sounds like func will show setup for button

File: AcademyApp/ViewControllers/LoginViewController.swift [hunk 5]
```diff
@@ -80,151 +85,131 @@ extension LoginViewController : UITextFieldDelegate
...
-// MARK: - Constants
+    func doneButtonSetup() {
+        doneButton.isEnabled = false
 
-private extension LoginViewController
-{
-    struct Config {
-        static let adjustedDoneButtonLoginFormVerticalSpacing : CGFloat = 56
+        doneButton.setTitle(doneButton.titleLabel?.text ?? "Done", for: .normal)
+        doneButton.titleLabel?.font = UIFont.doneButtonTitleFont
```
Comment (cejanen) [inline AcademyApp/ViewControllers/LoginViewController.swift]: UIFOnt & UIColor not needed here... if its obvious its not necessary to write class for enum etc

File: AcademyApp/ViewControllers/LoginViewController.swift [hunk 6]
```diff
@@ -10,68 +10,73 @@ import UIKit
...
 // MARK: - View Controller for login scree
 
-class LoginViewController: UIViewController
-{
+class LoginViewController: UIViewController {
+
+    // MARK: - Stored Properities
+
+    // default vertical spacing constraint constant between doneButton and login form
+    private var defaultVerticalSpacingConstant: CGFloat = 0
```
Comment (cejanen) [inline AcademyApp/ViewControllers/LoginViewController.swift]: Now you have one constant in enum config and second alone as property... Would be nice to have both in config and I would advice to rename config to UIConfig for example bc config is common name for anything :-D

**Review by cejanen — COMMENTED** [review]
Few little updates I would ask for thx :-D

File: .swiftlint.yml (hunk 3 above)
Comment (ZaweSK) [reply .swiftlint.yml:11]: Trailing_whitespace rule removed from disables section in .swiftlint.yaml

Comment (ZaweSK) [reply .swiftlint.yml]: Trailing_whitespace rule removed from disables section in .swiftlint.yaml

File: AcademyApp/ViewControllers/LoginViewController.swift (hunk 6 above)
Comment (ZaweSK) [reply AcademyApp/ViewControllers/LoginViewController.swift]: Renamed Config to UIConfig. Move all of the constants to UIConfig enum.

File: AcademyApp/ViewControllers/LoginViewController.swift (hunk 4 above)
Comment (ZaweSK) [reply AcademyApp/ViewControllers/LoginViewController.swift]: Renamed all setup methods to setup_______()

File: AcademyApp/ViewControllers/LoginViewController.swift (hunk 5 above)
Comment (ZaweSK) [reply AcademyApp/ViewControllers/LoginViewController.swift]: Removed unnecessary class names.

**Pull request merged**